File: 백준/Silver/9095. 1， 2， 3 더하기/1， 2， 3 더하기.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');
const T = Number(input[0]);
const MAX_N = 10;

const result = [];
const dp = new Array(MAX_N + 1).fill(0);
// 초기화
dp[1]=1; dp[2]=2; dp[3]=4;
for(let i=4; i<=MAX_N; i++) {
    dp[i] = dp[i-1] + dp[i-2] + dp[i-3];
}
// 계산된 값을 결과 배열에 넣기
for(let i=1; i<=T; i++) {
    const n = Number(input[i]);
    result.push(dp[n]);
}

console.log(result.join('\n'));

File: 백준/Silver/9095. 1， 2， 3 더하기/1， 2， 3 더하기.py
#  백준 9095 : 1, 2, 3 더하기
import sys
input = sys.stdin.readline

T = int(input().strip())
result = []

# 재귀함수 작성 : 1,2,3의 합으로 n을 만들 수 있는 경우의 수를 구하는 함수(dp 사용 X)
def solve_recursive(n):
    if n < 0:
        return 0
    if n == 0:
        return 1

    return solve_recursive(n - 1) + solve_recursive(n - 2) + solve_recursive(n - 3)

for _ in range(T):
    n = int(input().strip())
    result.append(solve_recursive(n))

sys.stdout.write('\n'.join(map(str, result)) + '\n')